### PR TITLE
CDAP-11527 add a Spark2 example application

### PR DESCRIPTION
**Pull request by albertshau**
Added an example application that trains a DecisionTree model
using Spark2. The app contains a service for loading training
data and retrieving information about models, and a Spark program
that trains new models and stores metadata about them.

### CODE REVIEW

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-DUT5770

File: cdap-examples/DecisionTreeRegression/pom.xml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Copyright © 2014 Cask Data, Inc.
```
Comment (awholegunch) [inline cdap-examples/DecisionTreeRegression/pom.xml]: 2017!

File: cdap-standalone/src/main/resources/cdap-site.xml [hunk 2]
```diff
@@ -53,6 +53,8 @@
     <value>spark1_2.10</value>
     <description>
       Spark compatibility version. Must be 'spark1_2.10' or 'spark2_2.11'.
+      This is only used in standalone mode. In distributed mode, the spark
```
Comment (awholegunch) [inline cdap-standalone/src/main/resources/cdap-site.xml]: the Spark

File: cdap-examples/DecisionTreeRegression/src/test/java/co/cask/cdap/examples/dtree/DecisionTreeRegressionAppTest.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Decision Tree tests.
```
Comment (awholegunch) [inline cdap-examples/DecisionTreeRegression/src/test/java/co/cask/cdap/examples/dtree/DecisionTreeRegressionAppTest.java]: Decision Tree Example tests.

File: cdap-examples/DecisionTreeRegression/src/main/java/co/cask/cdap/examples/dtree/ModelTrainer.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.examples.dtree;
+
+import co.cask.cdap.api.spark.AbstractSpark;
+
+/**
+ * Spark Client. Sets up the Main class.
```
Comment (awholegunch) [inline cdap-examples/DecisionTreeRegression/src/main/java/co/cask/cdap/examples/dtree/ModelTrainer.java]: Spark Client that sets up the Main class.

File: cdap-examples/DecisionTreeRegression/src/main/java/co/cask/cdap/examples/dtree/DecisionTreeRegressionApp.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import co.cask.cdap.api.data.schema.Schema;
+import co.cask.cdap.api.data.schema.UnsupportedTypeException;
+import co.cask.cdap.api.dataset.DatasetProperties;
+import co.cask.cdap.api.dataset.lib.FileSet;
+import co.cask.cdap.api.dataset.lib.ObjectMappedTable;
+import co.cask.cdap.api.dataset.lib.ObjectMappedTableProperties;
+
+/**
+ * Application that trains a decision tree regression model.
+ * Consists of a Service that is used to upload training data, and a model trainer Spark program.
```
Comment (awholegunch) [inline cdap-examples/DecisionTreeRegression/src/main/java/co/cask/cdap/examples/dtree/DecisionTreeRegressionApp.java]: Consists of a service used to upload training data and a Spark program model trainer .

**Comment by awholegunch** [comment]
A bunch of Javadoc comment-comments. I assume docs for this will follow in another PR?

File: cdap-examples/DecisionTreeRegression/pom.xml (hunk 1 above)
Comment (chtyim) [reply cdap-examples/DecisionTreeRegression/pom.xml]: Back to the future

File: cdap-examples/DecisionTreeRegression/src/main/scala/co/cask/cdap/examples/dtree/DecisionTreeRegressionTrainer.scala [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+
+    // Load the data stored in LIBSVM format as a DataFrame.
+    val data = spark.read.format("libsvm").load(inputPath.get())
+
+    // Automatically identify categorical features, and index them.
+    // Here, we treat features with > 4 distinct values as continuous.
+    val featureIndexer = new VectorIndexer()
+      .setInputCol("features")
+      .setOutputCol("indexedFeatures")
+      .setMaxCategories(4)
```
Comment (rohitsinha54) [inline cdap-examples/DecisionTreeRegression/src/main/scala/co/cask/cdap/examples/dtree/DecisionTreeRegressionTrainer.scala:65]: Should this be configurable ?

**Review by rohitsinha54 — APPROVED** [review]
one question. 

LGTM

File: cdap-examples/DecisionTreeRegression/src/main/scala/co/cask/cdap/examples/dtree/DecisionTreeRegressionTrainer.scala (hunk 6 above)
Comment (albertshau) [reply cdap-examples/DecisionTreeRegression/src/main/scala/co/cask/cdap/examples/dtree/DecisionTreeRegressionTrainer.scala:65]: all the params should probably be configurable, but can work on that later.

**Comment by albertshau** [comment]
https://builds.cask.co/browse/CDAP-DUT5770

**Pull request merged**